File: sdks/shrinkix-ruby/lib/shrinkix/resources/usage.rb
# frozen_string_literal: true

module Shrinkix
  module Resources
    # Usage Resource
    class Usage
      def initialize(transport)
        @transport = transport
      end

      def get_stats
        result = @transport.get('/usage/stats')
        result[:data].merge(rate_limit: result[:rate_limit])
      end
    end

    # Limits Resource
    class Limits
      def initialize(transport)
        @transport = transport
      end

      def get
        result = @transport.get('/limits')
        result[:data].merge(rate_limit: result[:rate_limit])
      end
    end

    # Validate Resource
    class Validate
      def initialize(transport)
        @transport = transport
      end

      def validate(file_size:, format:, width:, height:)
        result = @transport.post('/validate', body: {
          fileSize: file_size,
          format: format,
          width: width,
          height: height
        }.to_json)

        result[:data].merge(rate_limit: result[:rate_limit])
      end
    end
  end
end

File: sdks/shrinkix-php/src/Resources/Usage.php
<?php

namespace Shrinkix\Resources;

use Shrinkix\Transport;

/**
 * Usage Resource
 */
class Usage
{
    private Transport $transport;

    public function __construct(Transport $transport)
    {
        $this->transport = $transport;
    }

    public function getStats(): array
    {
        $result = $this->transport->get('/usage/stats');
        return array_merge($result['data'], ['rate_limit' => $result['rate_limit']]);
    }
}

/**
 * Limits Resource
 */
class Limits
{
    private Transport $transport;

    public function __construct(Transport $transport)
    {
        $this->transport = $transport;
    }

    public function get(): array
    {
        $result = $this->transport->get('/limits');
        return array_merge($result['data'], ['rate_limit' => $result['rate_limit']]);
    }
}

/**
 * Validate Resource
 */
class Validate
{
    private Transport $transport;

    public function __construct(Transport $transport)
    {
        $this->transport = $transport;
    }

    public function validate(int $fileSize, string $format, int $width, int $height): array
    {
        $result = $this->transport->post('/validate', [
            'json' => [
                'fileSize' => $fileSize,
                'format' => $format,
                'width' => $width,
                'height' => $height,
            ],
        ]);

        return array_merge($result['data'], ['rate_limit' => $result['rate_limit']]);
    }
}
